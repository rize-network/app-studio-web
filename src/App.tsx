import React, { Suspense } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { ComponentRouter } from 'src/routes/routes';
import { View, Text } from 'app-studio';

export const App: React.FC = () => {
  const location = useLocation();
  return (
    <Suspense
      fallback={
        <View
          height="100vh"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Text>Loading...</Text>
        </View>
      }
    >
      <Routes location={location}>
        {ComponentRouter.map((item, index) => (
          <Route key={index} path={item.path} element={item.element} />
        ))}
      </Routes>
    </Suspense>
  );
};

export default App;
