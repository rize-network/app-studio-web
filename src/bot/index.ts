import { Bot } from './Bot';
import { DocuCode } from './DocuCode';

async function main() {
  const projectName = 'MDX Documentation';
  const descriptionPath = 'docs/description.md';

  // Get componentName from command line arguments
  const componentName = process.argv[2]; // Assuming it's the first argument passed
  if (!componentName) {
    console.error('Please provide a component name as an argument.');
    process.exit(1);
  }

  console.log(`Generating documentation for component: ${componentName}`);

  const basePath = 'src/components';
  const propsPath = `src/data/props/${componentName}.props.json`;
  const componentFolder = `${basePath}/${componentName}`;
  const componentPath = `${basePath}/${componentName}/${componentName}`;

  const assistantGPT = new Bot(componentName);

  // Implementation for Documentation of the code
  console.log('Initializing documentation process...');
  const docuCode = new DocuCode(componentName);
  const assistantDocumentation = await assistantGPT.init(
    projectName,
    [],
    descriptionPath,
    'docu'
  );

  await docuCode.processDirectory(componentFolder, assistantDocumentation.id);

  // Generating props file for code
  console.log('Generating props file...');
  const { fileIds } = await assistantGPT.addFiles(componentPath);

  const assistantCreation = await assistantGPT.init(
    projectName,
    fileIds,
    descriptionPath,
    'props'
  );

  await assistantGPT.response(assistantCreation.id, propsPath);

  await assistantGPT.MarkdownGeneration(
    componentFolder,
    componentName,
    propsPath
  );

  console.log('Documentation generation completed.');
}

main().catch((error) => {
  console.error('Error during the execution of main:', error);
});
