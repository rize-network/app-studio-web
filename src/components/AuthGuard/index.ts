export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
