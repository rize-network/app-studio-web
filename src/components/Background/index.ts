export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
