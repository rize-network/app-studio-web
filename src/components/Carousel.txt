'use client';
import React, { useEffect, useRef, useState, createContext } from 'react';
import { Horizontal, Vertical } from './Layout';
import { IconButtons } from './Button/examples';
import { Button } from './Button/Button';
import { RightArrowSvg } from 'src/components/Svg/index';

// import { IconLeft, IconRight } from '@app-studio/web/icons';

interface CarouselProps {
  items: JSX.Element[];
  initialScroll?: number;
}

export const CarouselContext = createContext({
  onCardClose: (index: number) => {},
  currentIndex: 0,
});

export const Carousel = ({ items, initialScroll = 0 }: CarouselProps) => {
  const carouselRef = useRef<HTMLDivElement>(null);
  const [canScrollLeft, setCanScrollLeft] = useState(false);
  const [canScrollRight, setCanScrollRight] = useState(true);

  useEffect(() => {
    if (carouselRef.current) {
      carouselRef.current.scrollLeft = initialScroll;
      checkScrollability();
    }
  }, [initialScroll]);

  const checkScrollability = () => {
    if (carouselRef.current) {
      const { scrollLeft, scrollWidth, clientWidth } = carouselRef.current;
      setCanScrollLeft(scrollLeft > 0);
      setCanScrollRight(scrollLeft < scrollWidth - clientWidth);
    }
  };

  const scrollLeft = () =>
    carouselRef.current?.scrollBy({ left: -300, behavior: 'smooth' });
  const scrollRight = () =>
    carouselRef.current?.scrollBy({ left: 300, behavior: 'smooth' });

  return (
    <CarouselContext.Provider
      value={{ onCardClose: () => {}, currentIndex: 0 }}
    >
      <Horizontal
        gap={2}
        alignItems="center"
        width="100%"
        paddingVertical={20}
        position="relative"
      >
        <Button
          onClick={scrollLeft}
          icon={<RightArrowSvg />}
          disabled={!canScrollLeft}
          shape="rounded"
        />
        <Horizontal
          ref={carouselRef}
          onScroll={checkScrollability}
          gap={4}
          overflowX="auto"
          paddingX={10}
        >
          {items.map((item, index) => (
            <Vertical key={index} minWidth="300px">
              {item}
            </Vertical>
          ))}
        </Horizontal>
        <Button
          onClick={scrollRight}
          icon={<RightArrowSvg />}
          disabled={!canScrollRight}
          shape="rounded"
        />
      </Horizontal>
    </CarouselContext.Provider>
  );
};
