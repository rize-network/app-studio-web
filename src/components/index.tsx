export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
