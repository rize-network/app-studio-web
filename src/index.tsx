import React from 'react';
import ReactDOM from 'react-dom/client';
import AppProvider from 'src/providers';
import { App } from './App';
import { ComboBoxProvider } from './components/Form/ComboBox/ComboBox/ComboBox.provider';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <AppProvider>
      <ComboBoxProvider>
        <App />
      </ComboBoxProvider>
    </AppProvider>
  </React.StrictMode>
);
