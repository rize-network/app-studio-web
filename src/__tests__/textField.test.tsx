import React from 'react';
import renderer from 'react-test-renderer';
import { TextField } from 'src/components';
import { ProfileIcon } from 'src/components/Icon/Icon';

import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

afterEach(() => {
  cleanup();
});

test('renders TextField component', () => {
  render(<TextField name="username" role="textbox" />);
  const textFieldElement = screen.getByRole('textbox');
  expect(textFieldElement).toBeInTheDocument();
});

test('renders TextField with specified role', () => {
  render(<TextField name="username" role="textbox" />);
  const inputElement = screen.getByRole('textbox');
  expect(inputElement).toBeInTheDocument();
});

test('renders with placeholder', () => {
  render(<TextField name="username" placeholder="Enter your name" />);
  const inputElement = screen.getByPlaceholderText('Enter your name');
  expect(inputElement).toBeInTheDocument();
});

test('renders with initial value', () => {
  render(<TextField name="username" value="John" role="textbox" />);
  const inputElement = screen.getByRole('textbox');
  expect(inputElement).toHaveValue('John');
});

test('renders disabled TextField', () => {
  render(<TextField name="username" isDisabled />);
  const inputElement = screen.getByRole('textbox');
  expect(inputElement).toBeDisabled();
});

test('renders readonly TextField', () => {
  render(<TextField name="username" value="readonly value" isReadOnly />);
  const inputElement = screen.getByRole('textbox');
  expect(inputElement).toHaveAttribute('readonly');
  expect(inputElement).toHaveValue('readonly value');
  userEvent.type(inputElement, 'attempt to change');
  expect(inputElement).toHaveValue('readonly value');
});

// test('renders helper text when error is true', () => {
//   const helperText = 'This is helper text';
//   render(<TextField name="error" helperText={helperText} error={true} />);
//   const helperTextElement = screen.queryByText(helperText);
//   expect(helperTextElement).toBeInTheDocument();
// });

test('TextField to match snapshot', () => {
  const tree = renderer
    .create(
      <TextField
        name="name"
        placeholder="Name"
        left={<ProfileIcon size={12} />}
        error={true}
        helperText={'required'}
        variant="outline"
        size="xl"
        views={{
          box: {
            backgroundColor: 'red',
          },
          label: {
            fontSize: 15,
          },
          field: {
            backgroundColor: 'red',
          },
        }}
        isAutoFocus
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
