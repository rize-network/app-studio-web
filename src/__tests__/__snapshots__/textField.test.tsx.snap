// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField to match snapshot 1`] = `
<div
  className="sc-bdfCDU krrJWc"
  display="flex"
  flexDirection="column"
  flexWrap="nowrap"
  gap={5}
  justifyContent="flex-start"
  onClick={[Function]}
  position="relative"
>
  <div
    alignItems="center"
    backgroundColor="red"
    borderColor="error"
    borderRadius={0}
    borderStyle="solid"
    borderWidth={1}
    className="sc-bdfCDU krrJWc"
    color="theme.primary"
    cursor="pointer"
    display="flex"
    flexDirection="row"
    flexWrap="nowrap"
    fontSize={20}
    gap={10}
    justifyContent="space-between"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    paddingBottom={4}
    paddingLeft={16}
    paddingRight={36}
    paddingTop={10}
    transition="padding 0.2s ease"
    width="100%"
  >
    <div
      alignItems="center"
      className="sc-bdfCDU gmVaNF"
      display="flex"
      height="12px"
      justifyContent="center"
      onClick={[Function]}
      width="12px"
    >
      <svg
        fill="#000000"
        height="12px"
        version="1.1"
        viewBox="0 0 20 20"
        width="12px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="SVGRepo_bgCarrier"
          strokeWidth="0"
        />
        <g
          id="SVGRepo_tracerCarrier"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <g
          id="SVGRepo_iconCarrier"
        >
          <title>
            profile_round [#1342]
          </title>
           
          <desc>
            Created with Sketch.
          </desc>
          <defs>
             
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
            id="Page-1"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="white"
              id="Dribbble-Light-Preview"
              transform="translate(-140.000000, -2159.000000)"
            >
              <g
                id="icons"
                transform="translate(56.000000, 160.000000)"
              >
                <path
                  d="M100.562548,2016.99998 L87.4381713,2016.99998 C86.7317804,2016.99998 86.2101535,2016.30298 86.4765813,2015.66198 C87.7127655,2012.69798 90.6169306,2010.99998 93.9998492,2010.99998 C97.3837885,2010.99998 100.287954,2012.69798 101.524138,2015.66198 C101.790566,2016.30298 101.268939,2016.99998 100.562548,2016.99998 M89.9166645,2004.99998 C89.9166645,2002.79398 91.7489936,2000.99998 93.9998492,2000.99998 C96.2517256,2000.99998 98.0830339,2002.79398 98.0830339,2004.99998 C98.0830339,2007.20598 96.2517256,2008.99998 93.9998492,2008.99998 C91.7489936,2008.99998 89.9166645,2007.20598 89.9166645,2004.99998 M103.955674,2016.63598 C103.213556,2013.27698 100.892265,2010.79798 97.837022,2009.67298 C99.4560048,2008.39598 100.400241,2006.33098 100.053171,2004.06998 C99.6509769,2001.44698 97.4235996,1999.34798 94.7348224,1999.04198 C91.0232075,1998.61898 87.8750721,2001.44898 87.8750721,2004.99998 C87.8750721,2006.88998 88.7692896,2008.57398 90.1636971,2009.67298 C87.1074334,2010.79798 84.7871636,2013.27698 84.044024,2016.63598 C83.7745338,2017.85698 84.7789973,2018.99998 86.0539717,2018.99998 L101.945727,2018.99998 C103.221722,2018.99998 104.226185,2017.85698 103.955674,2016.63598"
                  id="profile_round-[#1342]"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
    <div
      className="sc-bdfCDU krrJWc"
      display="flex"
      flexDirection="column"
      flexWrap="nowrap"
      justifyContent="flex-start"
      onClick={[Function]}
      width="100%"
    >
      <input
        autoComplete="off"
        autoFocus={true}
        backgroundColor="red"
        border="none"
        className="sc-bdfCDU ibMnua"
        color="color.blueGray.700"
        cursor="auto"
        disabled={false}
        fontSize={20}
        heigth="100%"
        lineHeight={20}
        margin={0}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        padddingHorizontal={0}
        paddingVertical={8}
        placeholder="Name"
        readOnly={false}
        type="text"
        value=""
        width="100%"
      />
    </div>
  </div>
  <div
    className="sc-bdfCDU krrJWc"
    color="theme.error"
    fontSize={10}
    fontStyle="normal"
    fontWeight={400}
    lineHeight={10}
    marginHorizontal={0}
    marginVertical={0}
    onClick={[Function]}
    role="text"
    textDecoration="none"
  />
</div>
`;
